src/lib/pipeline.ts

```diff
@@ -175,6 +177,14 @@ function simplifyMongoPipeline(mongoSteps: Array<MongoStep>): Array<MongoStep> {
 
   for (const step of mongoSteps.slice(1)) {
     if (step.$project !== undefined && lastStep.$project !== undefined) {
+      for (const key in step.$project) {
+        // We do not want to merge two $project with common keys
+        if (key in lastStep.$project) {
+          outputSteps.push(step);
+          lastStep = step;
+          continue
+        }
+      }
```

Comment:
Good catch! I think we should test this case specifically and even probably in a separate commit (or PR).